### ISSUE TITLE
Race-prone use of makedirs() in junitxml.py causes test failure

### ISSUE THREAD

**Issue by nyh** [pytest-dev/pytest#10694]
In https://github.com/scylladb/scylladb/issues/12637 we saw a case where a pytest invocation failed with a FileExistsError reported in junitxml.py:
```
  File "/usr/lib/python3.11/site-packages/_pytest/junitxml.py", line 646, in pytest_sessionfinish
    os.makedirs(dirname)
  File "<frozen os>", line 225, in makedirs
FileExistsError: [Errno 17] File exists: '/jenkins/workspace/releng/Scylla-CI/scylla/testlog/x86_64/dev/xml/cql-pytest.cassandra_tests/validation/entities'
```

It turns out that our test framework ran multiple `pytest` processes in parallel for different test files. This should be fine.  Some of these test files are in the same directory. Each of these pytest invocations ran the following code from junitxml.py:
```python
        if not os.path.isdir(dirname):
            os.makedirs(dirname)
```
But this is race-prone: Two parallel processes run this code, they may both discover that the directory doesn't yet exist, both try `makedirs` and one of them will fail.

The solution is easy - just replace those two lines by `os.makedirs(dirname, exist_ok=True)` - this will consider the previous existence of the directory to be ok. the `exist_ok=True` was added in Python 3.2 - if for some reason you don't want to do this, you should at least catch the FileExistsError exception and ignore it.

**Comment by The-Compiler** [comment]
Duplicate of #10604 and already fixed in 7.2.1
